src/test/regress/expected/with.out

```diff
@@ -2207,3 +2207,48 @@ select count(*) from rank_tbl where rank in (select rank from updated);
      9
 (1 row)
 
+-- Test shared scan over values with singleQE join
+set optimizer = off;
+--start_ignore
+drop table if exists d;
+--end_ignore
+create table d (c1 int, c2 int) distributed by (c1);
+insert into d (values (2,0),(2,0));
+explain (costs off)
+with cte as (
+    select count(*) c1 from (values (1,2),(3,4)) v
+) select * from cte a join (select * from d join cte using(c1) limit 1) b using(c1);
+                                    QUERY PLAN                                    
+----------------------------------------------------------------------------------
+ Hash Join
+   Hash Cond: (b.c1 = share0_ref1.c1)
+   ->  Subquery Scan on b
+         ->  Limit
+               ->  Gather Motion 3:1  (slice2; segments: 3)
+                     ->  Limit
+                           ->  Hash Join
+                                 Hash Cond: (d.c1 = share0_ref2.c1)
+                                 ->  Seq Scan on d
+                                 ->  Hash
+                                       ->  Redistribute Motion 1:3  (slice1)
+                                             Hash Key: share0_ref2.c1
+                                             ->  Shared Scan (share slice:id 1:0)
+   ->  Hash
+         ->  Shared Scan (share slice:id 0:0)
+               ->  Materialize
+                     ->  Aggregate
+                           ->  Values Scan on "*VALUES*"
```

Comment:
This part is still executed on coordinator?

Reply:
yes

Reply:
>which led to different parts of SharedScan ending up on different segments,
which leads to deadlock.

Could you explain, please, the statement from PR description? Shared scans generally capable for communicate between QD and QE? Otherwise, why this case works?

Reply:
This part is executed on entry-db (which is same as coordinator) and in slice1 is redistributed to all segments